apps/test/unit/componentLibrary/ChipsTest.tsx

```diff
@@ -0,0 +1,126 @@
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React, {useState} from 'react';
+import '@testing-library/jest-dom';
+
+import Chips, {ChipsProps} from '@cdo/apps/componentLibrary/chips';
+
+const options = [
+  {value: 'chip1', label: 'Chip1'},
+  {value: 'chip2', label: 'Chip2'},
+  {value: 'chip3', label: 'Chip3'},
+];
+
+describe('Design System - Chips', () => {
+  const TestChips: React.FC<Partial<ChipsProps>> = props => {
+    const [values, setValues] = useState<string[]>(props.values || []);
+
+    return (
+      <Chips
+        name="test-chips"
+        values={values}
+        setValues={setValues}
+        options={options}
+        {...props}
+      />
+    );
+  };
+
+  it('renders with correct label', () => {
+    render(<TestChips label="Chips label" />);
+
+    expect(screen.getByText('Chips label')).toBeInTheDocument();
+    options.forEach(({label}) => {
+      expect(screen.getByLabelText(label)).toBeInTheDocument();
+    });
+  });
+
+  it('does not render label if it is not specified', () => {
+    render(<TestChips />);
+
+    expect(screen.queryByText('Chips label')).not.toBeInTheDocument();
+    options.forEach(({label}) => {
+      expect(screen.getByLabelText(label)).toBeInTheDocument();
+    });
+  });
+
+  it('checks chip on click', async () => {
+    const user = userEvent.setup();
+
+    render(<TestChips label="Chips label" />);
+
+    const chip1 = screen.getByLabelText('Chip1') as HTMLInputElement;
+    const chip2 = screen.getByLabelText('Chip2') as HTMLInputElement;
+    const chip3 = screen.getByLabelText('Chip3') as HTMLInputElement;
+
+    expect(chip1.checked).toBe(false);
+    expect(chip2.checked).toBe(false);
+    expect(chip3.checked).toBe(false);
+
+    await user.click(chip1);
+
+    expect(chip1.checked).toBe(true);
+    expect(chip2.checked).toBe(false);
+    expect(chip3.checked).toBe(false);
+
+    await user.click(chip2);
+
+    expect(chip1.checked).toBe(true);
+    expect(chip2.checked).toBe(true);
+    expect(chip3.checked).toBe(false);
+
+    await user.click(chip1);
+
+    expect(chip1.checked).toBe(false);
+    expect(chip2.checked).toBe(true);
+    expect(chip3.checked).toBe(false);
+  });
+
+  it('handles required state correctly', async () => {
```

Comment:
This test confused me a bit (ie, the expected behavior is that clicking a single chip flips the required state of all chips). I understand why this is the case since I've been in this code recently for the i18n work, but I think it could be confusing to someone new to it. Could you add a comment to give a bit of context?

Reply:
will update and then merge

Reply:
Ok. turned out there was a comment already[ in the Chips component,](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/componentLibrary/chips/Chips.tsx#L102) updated it slightly and added copy of that comment to test file as well